### ISSUE TITLE
[<firebase_in_app_messaging>] <android only shows repeating campaigns at the first time>

### ISSUE THREAD

**Issue by nsndimt** [FirebaseExtended/flutterfire#2082]
**Describe the bug**
The same campaign with the displaying interval set to once per day can be properly displayed in ios while android only shows it at the first time.

**To Reproduce**
1. go to firebase console, create two identical campaign(image-only style), one for android, one for ios
2. set the campaign's displaying interval to be once per day
3. open app on an android phone, see the pop-up at the first time.  After that, fiam outputs "already impressed xxx ? : true" to log at each opening and stop showing the pop-up.

**Expected behavior**
android correctly show the pop up every day

**Additional context**
flutter doctor output:
```
/Users/nsndimt/development/flutter/bin/flutter doctor --verbose
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale en-US)
    • Flutter version 1.12.13+hotfix.8 at /Users/nsndimt/development/flutter
    • Framework revision 0b8abb4724 (2 weeks ago), 2020-02-11 11:44:36 -0800
    • Engine revision e1e6ced81d
    • Dart version 2.7.0

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/nsndimt/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 43.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[!] IntelliJ IDEA Ultimate Edition (version 2019.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.
Process finished with exit code 0
```
package version:
```
firebase_core: ^0.4.4
firebase_analytics: ^5.0.11
firebase_in_app_messaging: ^0.1.1+1
```
**Possible Cause**

in [ios code](https://github.com/firebase/firebase-ios-sdk/blob/48aa5ed5ee30d01a184e757df806169195df00df/FirebaseInAppMessaging/Sources/Private/Flows/FIRIAMBookKeeper.h#L52) I can find a comment that shows firebase ios package has taken repeating campaigns into consideration.

in [android code](https://github.com/firebase/firebase-android-sdk/blob/efd6b2770c6b08e9534baa92ca8ea4cd76310672/firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InAppMessageStreamManager.java#L185) it seems that all campaigns that have been shown before will be filtered.

So, my theory is that Firebase Android SDK forgets to treat repeating campaigns differently. But I am a rookie with regard to android and ios development. My theory can be totally wrong.  I just want to help.

**Comment by iapicca** [comment]
Hi @nsndimt 
can you please provide your `flutter run --verbose` 
and your `pubspec.yaml` 
Thank you

**Comment by nsndimt** [comment]
I notice there's an update. So I update my firebase-in-app-messaging package to 0.1.1+3. 
The original campaign has ended. I create a test campaign that will only be triggered by the "test_show_pop_up" event. I manually call```triggerEvent('test_show_pop_up')``` inside my homepage's ```void initState() {...}``` function. 
Previously, I just added ```firebase_in_app_messaging: ^0.1.1+1``` to my ```pubspec.yaml``` I didn't create an instance of ```FirebaseInAppMessaging``` by calling ```  final firebaseInAppMessaging = FirebaseInAppMessaging();``` I also didn't import ```firebase_in_app_messaging.dart```  This time I import the header and create one instance of ```FirebaseInAppMessaging```. I do not know if this will make a difference. 

I have deleted my app and re-installed it. 

At the first time, the message shows as expected.  But I still need to wait serval days to see if the problem exists.
